src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,74 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useDispatch } from 'react-redux';
+import { useAppSelector } from '../../app/hook';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { clearTodo } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const dispatch = useDispatch();
+  const [loading, setLoading] = useState<boolean>(false);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    setLoading(true);
+    if (typeof currentTodo?.userId === 'number') {
+      getUser(currentTodo?.userId)
+        .then(db => setUser(db))
+        .finally(() => setLoading(false));
+    }
+  }, [currentTodo]);
+
+  const hndaleClearCurrentTask = () => {
```

Comment:
There is a typo in the function name `hndaleClearCurrentTask`. It should be corrected to `handleClearCurrentTask` to follow standard naming conventions.